rktlet/runtime/pod_sandbox.go

```diff
@@ -0,0 +1,246 @@
+
+	var rktUUID string
+	// TODO, inotify watch for the uuid file would be slightly more efficient
+	// We could also create a pair of pipes for this
+	for i := 0; i < 100; i++ {
+		data, err := ioutil.ReadAll(podUUIDFile)
+		if err != nil {
+			return nil, fmt.Errorf("error reading rkt pod UUID file: %v", err)
+		}
+		if len(data) != 0 {
+			rktUUID = string(data)
+			break
+		}
+
+		time.Sleep(100 * time.Millisecond)
+	}
+
+	if rktUUID == "" {
+		return nil, fmt.Errorf("waited 10s for pod sandbox to start, but it didn't: %v", k8sPodUid)
+	}
+
+	return &runtimeApi.RunPodSandboxResponse{
+		PodSandboxId: &rktUUID,
+	}, nil
+}
+
+func (r *RktRuntime) StopPodSandbox(ctx context.Context, req *runtimeApi.StopPodSandboxRequest) (*runtimeApi.StopPodSandboxResponse, error) {
+	respLines, err := r.RunCommand("stop", req.GetPodSandboxId())
+	// TODO, structured output will be so much nicer!
+	for _, line := range respLines {
+		if strings.HasSuffix(line, "is not running") {
+			return &runtimeApi.StopPodSandboxResponse{}, nil
+		}
+	}
+	return nil, err
+}
+
+func (r *RktRuntime) RemovePodSandbox(ctx context.Context, req *runtimeApi.RemovePodSandboxRequest) (*runtimeApi.RemovePodSandboxResponse, error) {
+	if _, err := r.RunCommand("rm", req.GetPodSandboxId()); err != nil {
+		return nil, err
+	}
+	return &runtimeApi.RemovePodSandboxResponse{}, nil
+}
+
+func (r *RktRuntime) PodSandboxStatus(ctx context.Context, req *runtimeApi.PodSandboxStatusRequest) (*runtimeApi.PodSandboxStatusResponse, error) {
+	resp, err := r.RunCommand("status", req.GetPodSandboxId())
+	if err != nil {
+		return nil, err
+	}
+
+	rktStatus := parseRktStatus(resp)
+	apiStatus := runtimeApi.PodSandBoxState_NOTREADY
+	var ip string
+	if rktStatus.state == "running" {
+		apiStatus = runtimeApi.PodSandBoxState_READY
+		ip = parseRktNetworkIp(rktStatus.network)
+	}
+
+	createdUnix := rktStatus.created.Unix()
+	return &runtimeApi.PodSandboxStatusResponse{
+		Status: &runtimeApi.PodSandboxStatus{
+			Id:          req.PodSandboxId,
+			Metadata:    nil, // TODO
+			State:       &apiStatus,
+			CreatedAt:   &createdUnix,
+			Network:     &runtimeApi.PodSandboxNetworkStatus{Ip: &ip},
+			Linux:       nil, // TODO
+			Labels:      nil, // TODO
+			Annotations: nil, // TODO
+		},
+	}, nil
+}
+
+type rktStatusResponse struct {
+	state   string
+	created time.Time
+	network string
+}
+
+func parseRktStatus(status []string) rktStatusResponse {
+	resp := rktStatusResponse{}
+
+	for _, line := range status {
+		parts := strings.SplitN(line, "=", 2)
+		if len(parts) != 2 {
+			glog.Warningf("malformed rkt status response: expected an '=', got: %v", line)
+			continue
+		}
+
+		switch parts[0] {
+		case "state":
+			resp.state = parts[1]
+		case "created":
+			time, err := time.Parse("2006-01-02 15:04:05.999 -0700 MST", parts[1])
+			if err != nil {
+				glog.Warningf("error parsing rkt started time: %v", err)
+			}
+			resp.created = time
+		case "networks":
+			resp.network = parts[1]
```

Comment:
@pskrzyns For this piece of code, we assume there could be a string that contains multiple networks. e.g. (`networks=default:ip4=172.16.28.8,foo:ip4=10.10.10.4`)
And we handle this by `parseRktNetworkIp()`